command/proxy_test.go

```diff
@@ -1156,6 +1156,178 @@ log_level = "trace"
 	wg.Wait()
 }
 
+// TestProxy_Cache_StaticSecretPermissionsLost Tests that the cache successfully caches a static secret
+// going through the Proxy for a KVV2 secret, and then the calling client loses permissions to the secret,
+// so it can no longer access the cache.
+func TestProxy_Cache_StaticSecretPermissionsLost(t *testing.T) {
+	logger := logging.NewVaultLogger(hclog.Trace)
+	cluster := vault.NewTestCluster(t, &vault.CoreConfig{
+		LogicalBackends: map[string]logical.Factory{
+			"kv": logicalKv.VersionedKVFactory,
+		},
+	}, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+
+	serverClient := cluster.Cores[0].Client
+
+	// Unset the environment variable so that proxy picks up the right test
+	// cluster address
+	defer os.Setenv(api.EnvVaultAddress, os.Getenv(api.EnvVaultAddress))
+	os.Unsetenv(api.EnvVaultAddress)
```

Comment:
🤔 I think this would work and save us a line:
```suggestion
	t.Setenv(api.EnvVaultAddress, "")
```

Reply:
Hmmm, I think I tried this once before for an Agent/Proxy test and there was a reason it didn't work, so I'll leave this for now, especially since that way we're consistent.